### ISSUE TITLE
how to detect duplicate nebula IPs

### ISSUE THREAD

**Issue by theblop** [slackhq/nebula#624]
I recently had a problem that took me a while to debug: I mistakenly configured 2 nebula certs with the same IP and this (unsurprisingly) resulted in random connection problems to applications running on these servers.

Is there a way to detect if the same IP is configured on more than one node from inside the mesh?
Maybe a nebula network summary view from the command line?

**Comment by rucciva** [comment]
Hi, not a direct solution, what do you think of using [terraform provider](https://registry.terraform.io/providers/TelkomIndonesia/nebula) for managing your entire cluster certificate? A [terraform module](https://registry.terraform.io/modules/TelkomIndonesia/mesh/nebula) is also available for quick setup. The module will throw error if duplicate IPs is detected. Examples are [here](https://github.com/TelkomIndonesia/terraform-nebula-mesh/blob/master/examples)

**Comment by theblop** [comment]
> The module will throw error if duplicate IPs is detected in the input variable. Examples of input variables are [here](https://github.com/TelkomIndonesia/terraform-nebula-mesh/blob/master/examples)

Thank you for pointing me towards this new terraform module! I already have a script to generate certs that checks for duplicate IPs but the terraform module will be much better than my script.

It doesn't really solve my problem though, in my case someone also recently cloned a VM that already contained a cert to test something... So ideally I'd like to check for duplicates with our monitoring/alerting tool.

**Comment by futurecad** [comment]
> > The module will throw error if duplicate IPs is detected in the input variable. Examples of input variables are [here](https://github.com/TelkomIndonesia/terraform-nebula-mesh/blob/master/examples)
> 
> Thank you for pointing me towards this new terraform module! I already have a script to generate certs that checks for duplicate IPs but the terraform module will be much better than my script.
> 
> It doesn't really solve my problem though, in my case someone also recently cloned a VM that already contained a cert to test something... So ideally I'd like to check for duplicates with our monitoring/alerting tool.


Has this issue been resolved?

**Comment by theblop** [comment]
> > > The module will throw error if duplicate IPs is detected in the input variable. Examples of input variables are [here](https://github.com/TelkomIndonesia/terraform-nebula-mesh/blob/master/examples)
> > 
> > 
> > Thank you for pointing me towards this new terraform module! I already have a script to generate certs that checks for duplicate IPs but the terraform module will be much better than my script.
> > It doesn't really solve my problem though, in my case someone also recently cloned a VM that already contained a cert to test something... So ideally I'd like to check for duplicates with our monitoring/alerting tool.
> 
> Has this issue been resolved?

No, there is nothing preventing 2 nodes connecting with the same certificate.

I understand now that it's probably impossible to fix... How could a lighthouse detect that a client is reusing the same cert as a previous one? The only thing that would be different would be the client public IP, but that can just be the old client changing its IP...

It's similar to the duplicate IPs problem in a LAN, but there is no ARP broadcast in the nebula layer :)
